### ISSUE TITLE
Allow definition of embedded DSL like asm and sql

### ISSUE THREAD

**Issue by decaruju** [vlang/v#14092]
Currently `asm` and `sql` are handled at the langage level, which means that adding a language to be used like sql is not possible.
```
sql db {
	insert new_customer into Customer
}
```
The same behaviour could be possible with a function definition like this :
```
fn sql(db Database, query string) {
  db.cursor.execute(query)
}
```
This way the `sql` keyword becomes syntactic sugar for
```
sql(db, "insert new_customer into Customer")
```
Allowing the creation of arbitrary dsl would allow for user extensions which would be quite helpful, such as
```
data := json {
  "foo": "bar,
  "baz": 2
}
```
Or weirder use-cases
```
small_squares := python {
  [x for i in range(10) if x := i**2 < 50]
}
```

**Comment by JalonSolov** [comment]
No, V will never allow/support DSL.  It is too error-prone, and too hard to support.  Alex has said this before, and I haven't heard anything about him changing his mind.

Personally, I agree with him.

@medvednikov ?

**Comment by decaruju** [comment]
When I was reading the doc I was wondering "how is the sql orm implemented", until I found out it is supported in the language directly. 

I might not have been clear as to how I propose supporting DSLs, simply as syntatic sugar over function calls, all code inside the braces is passed as a raw string :
```
fn raw_json(json_string string) T {
  // Adding braces to have valid json
  return json.decode("{$json_string}")
}
data := raw_json {
    "foo": 1,
    "bar": 2
}
```

I understand that it still adds complexity and that support could be problematic, but i think it makes for a much cleaner syntax in some cases.
